### PR TITLE
feat: implement process_justification_and_finalization function

### LINKED ISSUE #254: [epoch processing] Implement `process_justification_and_finalization`
https://github.com/lambdaclass/lambda_ethereum_consensus/blob/66059de3653c17ae620ce2aca2db61c558057f6a/docs/specs/beacon-chain.md?plain=1#L2161-L2173

See Spec tests format: https://github.com/ethereum/consensus-specs/tree/dev/tests/formats/epoch_processing

Acceptance criteria:
- Run corresponding spec tests by removing https://github.com/lambdaclass/lambda_ethereum_consensus/blob/66059de3653c17ae620ce2aca2db61c558057f6a/test/spec/runners/epoch_processing.ex#L12.

### PR DESCRIPTION
**Pull request by h3lio5**
**Motivation**

implements process_justification_and_finalization() function

Closes #254

### CODE REVIEW

File: lib/lambda_ethereum_consensus/state_transition/accessors.ex [hunk 1]
```diff
@@ -121,4 +121,131 @@ defmodule LambdaEthereumConsensus.StateTransition.Accessors do
...
+  @doc """
+  Return the block root at a recent ``slot``
+  """
+  @spec get_block_root_at_slot(BeaconState.t(), SszTypes.slot()) ::
+          {:ok, SszTypes.root()} | {:error, binary()}
+  def get_block_root_at_slot(state, slot) do
+    slots_per_root = ChainSpec.get("SLOTS_PER_HISTORICAL_ROOT")
+
+    if slot < state.slot and state.slot <= slot + slots_per_root do
+      root = Enum.at(state.block_roots, rem(slot, slots_per_root))
```
[the remark this answers is not in the document]
Comment (h3lio5) [reply lib/lambda_ethereum_consensus/state_transition/accessors.ex:165]: do you mean BeaconState in `lib/ssz_types/beacon_chain/beacon_state.ex`?

File: lib/lambda_ethereum_consensus/state_transition/epoch_processing.ex [hunk 2]
```diff
@@ -252,4 +252,150 @@ defmodule LambdaEthereumConsensus.StateTransition.EpochProcessing do
...
+
+    if previous_epoch_target_balance * 3 >= total_active_balance * 2 do
+      new_checkpoint = %SszTypes.Checkpoint{
+        epoch: previous_epoch,
+        root: previous_block_root
+      }
+
+      # <<just_bits_binary::binary>> = state.justification_bits
+      # <<first_bit, second_bit, third_bit, fourth_bit>> = just_bits_binary
+      [first_bit, second_bit, third_bit, fourth_bit] = get_bits_from_binary(state.justification_bits)
```
[the remark this answers is not in the document]
Comment (h3lio5) [reply lib/lambda_ethereum_consensus/state_transition/epoch_processing.ex:332]: ah I see. In Elixir, you cannot mutate (or even read?) the values from the parent scope, unlike in other languages like rust that allow it?

[the remark this answers is not in the document]
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/state_transition/epoch_processing.ex]: You can read values from outer scopes, but not mutate them.

**Comment by mpaulucci** [comment]
seems this was replaced by https://github.com/lambdaclass/lambda_ethereum_consensus/pull/441

**Pull request closed without merging**